File: data/readable_format/cpp/file1022.cpp
#include <iostream>
#include <vector>
#include <cstddef> // for size_t

// Define the Dgefa class and its method in the same file
template<typename T = double>
class Dgefa {
public:
    // Assuming dgefa function sets INFO to the sum of diagonal elements of A for this example
    static void dgefa(std::vector<std::vector<T>>& A, const size_t LDA, const size_t N, std::vector<int>& IPVT, int& INFO) {
        INFO = 0;
        for (size_t i = 0; i < N; ++i) {
            INFO += A[i][i];
        }
    }
};

// Simple testing mechanism without external libraries
void testDgefa() {
    const size_t N = 4;
    std::vector<std::vector<double>> A(N, std::vector<double>(N, 0.0));
    std::vector<int> IPVT(N);
    int INFO = 0;

    // Initialize A with a simple test case
    for (size_t i = 0; i < N; ++i) {
        A[i][i] = static_cast<double>(i + 1); // Adjust for 0-based indexing
    }

    // Call the function
    Dgefa<>::dgefa(A, N, N, IPVT, INFO);

    // Check the result
    if (INFO == 10) {
        std::cout << "Test Passed: INFO is correct." << std::endl;
    } else {
        std::cout << "Test Failed: Expected INFO = 10, but got INFO = " << INFO << std::endl;
    }
}

int main() {
    testDgefa();
    return 0;
}

File: data/readable_format/fortran/file1022.f90
PROGRAM test_dgefa
  IMPLICIT NONE
  INTEGER, PARAMETER :: dp = KIND(1.0D0) ! Double precision kind
  INTEGER, PARAMETER :: N = 4, LDA = N
  REAL(dp), DIMENSION(LDA,N) :: A
  INTEGER, DIMENSION(N) :: IPVT
  INTEGER :: INFO, i

  ! Initialize A with a simple test case
  A = 0.0_dp
  DO i = 1, N
     A(i,i) = REAL(i, dp)
  END DO

  CALL dgefa(A, LDA, N, IPVT, INFO)

  ! Simple test: INFO should be the sum of diagonal elements of A
  IF (INFO == 10) THEN
     PRINT *, "Test Passed."
  ELSE
     PRINT *, "Test Failed. INFO:", INFO
  END IF

CONTAINS

  SUBROUTINE dgefa(A, LDA, N, IPVT, INFO)
    INTEGER, INTENT(IN) :: LDA, N
    REAL(dp), DIMENSION(LDA,N), INTENT(INOUT) :: A
    INTEGER, DIMENSION(N), INTENT(OUT) :: IPVT
    INTEGER, INTENT(OUT) :: INFO
    
    ! Dummy implementation - setting INFO as sum of diagonal elements for testing
    INTEGER :: i
    INFO = 0
    DO i = 1, N
       INFO = INFO + A(i, i)
    END DO
  END SUBROUTINE dgefa

END PROGRAM test_dgefa